runtime/util/hshelp.c

```diff
@@ -2641,6 +2642,13 @@ recreateRAMClasses(J9VMThread * currentThread, J9HashTable * classHashTable, J9H
 		/* Delete original class from defining loader's class table */
 		if (!fastHCR) {
 			vmFuncs->hashClassTableDelete(classLoader, J9UTF8_DATA(className), J9UTF8_LENGTH(className));
+			if ((NULL != vm->sharedClassConfig) && (NULL != vm->sharedClassConfig->romToRamHashTable)) {
+				omrthread_rwmutex_enter_write(vm->sharedClassConfig->romToRamHashTableMutex);
+				RomToRamEntry entry;
```

Comment:
This isn't strictly legal C code: declarations must be at the beginning of a block.

Moving the initialization of `entry` before acquiring the mutex also slightly reduces the size of that critical section.

Reply:
Thanks Keith, I will open a pull request to update this.